src/affine.rs

```diff
@@ -198,10 +198,10 @@ mod tests {
         assert_eq!(
             affine_to_quaternion(&affine),
             RowVector4::new(
```

Comment:
Well, this one could really use an approximate equality fix, so as to avoid having test failures due to floating point arithmetic precision issues. By tapping into `RowVector4`, we can obtain a slice, and from there employ one of the macros in the `approx` crate.

Here's a patch that you can apply to address this:

```diff
diff --git a/src/affine.rs b/src/affine.rs
index e158519..cebe1da 100644
--- a/src/affine.rs
+++ b/src/affine.rs
@@ -195,14 +195,17 @@ mod tests {
         );
 
         let affine = Matrix3::new(1.1, 0.1, 0.1, 0.2, 1.1, 0.5, 0.0, 0.0, 1.0);
-        assert_eq!(
-            affine_to_quaternion(&affine),
-            RowVector4::new(
+
+        let quaternion = affine_to_quaternion(&affine);
+        assert_abs_diff_eq!(
+            quaternion.as_slice(),
+            &[
                 0.9929998817020889,
                 -0.11474227051531193,
                 0.017766153114299018,
-                0.02167510323267152
-            )
+                0.02167510323267152,
+            ][..],
+            epsilon = 1e-11,
         );
     }
 
diff --git a/src/lib.rs b/src/lib.rs
index 5724dc7..05c612f 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -55,6 +55,7 @@
 #[cfg(feature = "nalgebra_affine")] extern crate alga;
 #[cfg(feature = "nalgebra_affine")] extern crate nalgebra;
 #[cfg(feature = "ndarray_volumes")] extern crate ndarray;
+#[cfg(all(test, feature = "nalgebra_affine"))] #[macro_use] extern crate approx;
 
 extern crate byteordered;
 extern crate flate2;
```

Reply:
That pretty nice, thanks.